folly/portability/Windows.h

```diff
@@ -37,6 +37,31 @@
 #include <direct.h> // nolint
 #endif
 
+#ifndef _WINDOWS_
+// We're building in C++ 17 mode, but certain dependencies (Boost) still have
+// dependencies on unary_function and binary_function, so we have to make sure
+// not to remove them.
```

Comment:
`NOMINMAX`, `WIN32_LEAN_AND_MEAN` and `_WINSOCK_DEPRECATED_NO_WARNINGS` are the only defines that can live here. The rest of the defines need to stay where they are, as they effect things other than the headers that this includes.

If either `_WINDOWS_` or `_WINSOCK2API_` is defined, this needs to be a hard `#error`. Folly won't work if `NOMINMAX` isn't defined.